augurpy/calculate_AUC.py

```diff
@@ -71,39 +75,110 @@ def draw_subsample(input: DataFrame, augur_mode: str, subsample_size: int, strat
             permuting the labels
         subsample_size: number of cells to subsample randomly per type from each experimental condition
         stratified: if `True` subsamples are stratified according to condition
+        random_state: set random seed
 
     Returns:
         Subsample of input of size subsample_size
     """
-    pass
+    if augur_mode == "permut":
+        # shuffle labels
+        y_columns = [col for col in adata.obs if col.startswith("y")]
+        adata.obs[y_columns] = adata.obs[y_columns].sample(frac=1, random_state=random_state).reset_index(drop=True)
 
+    if augur_mode == "velocity":
+        # no feature selection, assuming this has already happenend in calculating velocity
+        subsample = sc.pp.subsample(adata, n_obs=subsample_size, copy=True, random_state=random_state)
+        return subsample
 
-def subsample_cross_validate(
-    input: DataFrame,
-    augur_mode: str,
-    subsample_size: int,
-    stratified: bool,
-    estimator: Union[RandomForestRegressor, RandomForestClassifier, LogisticRegression],
-    metrics: list,
-    folds: int,
-    subsample_idx: int,
-) -> DataFrame:
-    """Cross validate subsample.
+    # randomly sample features
+    # random.seed()
+    print(random, type(random))
+    features = sample(adata.var_names.tolist(), floor(len(adata.var_names.tolist()) * feature_perc))
+    # randomly sample samples
+    subsample = sc.pp.subsample(adata[:, features], n_obs=subsample_size, copy=True, random_state=random_state)
+    return subsample
+
+
+def average_metrics(subsample_cv_results: List[Any]) -> Dict[Any, Any]:
+    """Calculate average scores across multiple cross validation runs.
 
     Args:
-        input: Pandas DataFrame
+        cross_validation_results: list of all subsample cross validations
+
+    Returns:
+        Dict containing the average result for each metric.
+    """
+    metric_names = [m for m in [*subsample_cv_results[0].keys()] if m.startswith("mean")]
```

Comment:
```suggestion
    metric_names = [metric for metric in [*subsample_cv_results[0].keys()] if metric.startswith("mean")]
```

As long as it stays readable always prefer explicit over short variable names.